worlds/stardew_valley/regions.py

```diff
@@ -353,8 +354,81 @@ def inverted(self):
 ]
 
 
+# Attempt to implement remote loading of modded data if relevant (to avoid circular references and clutter)
+def loaded_mod_region_data(mod: str, data_type: str, world_options: StardewOptions):
+    loaded_mod_data = []
+    if mod in world_options[options.Mods]:
+        for mod_data in ModDataList:
+            if (mod_data.mod_name == mod) & (data_type == "Regions"):
+                for region in mod_data.regions:
+                    if not mod_data.regions[region]:
+                        loaded_mod_data.append(RegionData(region))
+                        continue
+                    loaded_mod_data.append(RegionData(region, mod_data.regions[region]))
+            if (mod_data.mod_name == mod) & (data_type == "Connections"):
+                for entrance in mod_data.connections:
+                    if mod_data.flags[entrance] == "NON_RANDOMIZED":
+                        loaded_mod_data.append(ConnectionData(entrance, mod_data.connections[entrance],
+                                                              flag=RandomizationFlag.NOT_RANDOMIZED))
+                    if mod_data.flags[entrance] == "NON_PROGRESSION":
+                        loaded_mod_data.append(ConnectionData(entrance, mod_data.connections[entrance],
+                                                              flag=RandomizationFlag.NON_PROGRESSION))
+                    if mod_data.flags[entrance] == "PELICAN_TOWN":
+                        loaded_mod_data.append(ConnectionData(entrance, mod_data.connections[entrance],
+                                                              flag=RandomizationFlag.PELICAN_TOWN))
+                    if mod_data.flags[entrance] == "BUILDINGS":
+                        loaded_mod_data.append(ConnectionData(entrance, mod_data.connections[entrance],
+                                                              flag=RandomizationFlag.BUILDINGS))
+                    if mod_data.flags[entrance] == "EVERYTHING":
+                        loaded_mod_data.append(ConnectionData(entrance, mod_data.connections[entrance],
+                                                              flag=RandomizationFlag.EVERYTHING))
+            if (mod_data.mod_name == mod) & (data_type == "Region Remover"):
+                for name in mod_data.region_remover:
+                    loaded_mod_data.append(RegionRemoverData(name, mod_data.region_remover[name]))
+            if (mod_data.mod_name == mod) & (data_type == "Connection Remover"):
+                for name in mod_data.connection_remover:
+                    loaded_mod_data.append(ConnectionRemoverData(name, mod_data.region_remover[name]))
+    return loaded_mod_data
+
+
+def region_extend(mod: str, existing_regions: List[Region], new_regions: List[Region], world_options: StardewOptions):
```

Comment:
I have a lot of things to say about this method, but honestly, if you apply the other changes I've requested in this file, this whole method will either disappear, or shrink significantly and I believe everything that is wrong with it will disappear organically. So I don't really need to say anything, other than "don't hesitate to completely gut this method. It will probably be the correct choice"

Reply:
I've continued to use a similar method since we need a means of merging regions together, but its been changed into something else.